src/main/java/de/uni_tuebingen/ub/nppm/db/SaveHelper.java

```diff
@@ -47,6 +50,57 @@ public static List<Map<String, String>> selectAttribute(String table, String att
         return results;
     }
 
+    public static List<Map> selectAttributeMap(String table, String attributeString, int id) throws Exception {
+        List<Map> results = new ArrayList<>();
+
+        try ( Session session = getSession()) {
+            session.getTransaction().begin();
+            String sql = "SELECT " + attributeString + " FROM " + table + " WHERE ID = " + id; // SQL-Abfrage erstellen
+
+            NativeQuery query = session.createNativeQuery(sql);
+
+            List<Object[]> queryResults = query.getResultList(); // Ergebnisse aus der Datenbank als Liste von Object-Arrays
+
+            for (Object[] row : queryResults) {
+                Map<String, Object> rowMap = new HashMap<>();
+
+                // Die Spaltenwerte in die Map einfügen, wobei die Spaltennamen als Schlüssel verwendet werden
+                for (int i = 0; i < row.length; i++) {
+                    String columnName = attributeString.split(",")[i].trim(); // Die Spaltennamen sind mit Kommas getrennt
+                    rowMap.put(columnName, row[i]);
+                }
+
+                results.add(rowMap);
+            }
+
+            session.getTransaction().commit();
+        }
+
+        return results;
+    }
+
+    public static List<Map<String, String>> getMapField(String zieltabelle, int id) throws Exception {
+        List<Map<String, String>> results = new ArrayList<>();
+
+        try ( Session session = getSession()) {
+            session.getTransaction().begin();
+            String sql = "SELECT * FROM " + zieltabelle + " WHERE ID = " + id; // SQL-Abfrage erstellen
+
+            SQLQuery query = session.createSQLQuery(sql);
+            query.setResultTransformer(Criteria.ALIAS_TO_ENTITY_MAP);
```

Comment:
Same here. You should not use the ResultTransformer directly since it is deprecated and we have to replace it later when migrating to Hibernate 6. Instead, just use `AbstractBase.getMappedList(String query)`, then you also have to write less code again.

Reply:
but the Abstract method have no return type of List<Map<String, String>>